php/Terminus/KLogger.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace Terminus;
+
+use DateTime;
+use Terminus\Runner;
+use Katzgrau\KLogger\Logger;
+use Psr\Log\LogLevel;
+
+//TODO: Change this classname to Logger once the old logger is fully replaced
+class KLogger extends Logger {
+
+  /**
+   * Class constructor. Feeds in output destination from env vars
+   *
+   * @param [array]  $options           Options for operation of logger
+   *        [array] config Configuration options from Runner
+   * @param [string] $logDirectory      File path to the logging directory
+   * @param [string] $logLevelThreshold The LogLevel Threshold
+   * @return [KLogger] $this
+   */
+  public function __construct(
+    array $options = array(),
+    $logDirectory = 'php://stdout',
+    $logLevelThreshold = LogLevel::INFO
+  ) {
+    $config = $options['config'];
+    unset($options['config']);
+
+    if (isset($_SERVER['TERMINUS_LOG_DIR'])) {
+      $logDirectory = $_SERVER['TERMINUS_LOG_DIR'];
+    } elseif ($config['silent']) {
+      $logDirectory = '/var/log/terminus/';
+    }
+
+    if ($config['debug']) {
+      $logLevelThreshold = LogLevel::DEBUG;
+    }
+
+    if ($config['json'] != null) {
+      $this->options['logFormat'] = 'json';
+    }
+    if ($config['bash'] != null) {
+      $this->options['logFormat'] = 'bash';
+    }
+
+    parent::__construct($logDirectory, $logLevelThreshold, $options);
+  }
+
+  /**
+    * Logs with an arbitrary level.
+    *
+    * @param [mixed]  $level
+    * @param [string] $message
+    * @param [array]  $context
+    * @return [void]
+    */
+  public function log($level, $message, array $context = array()) {
+    if ($this->logLevels[$this->logLevelThreshold] < $this->logLevels[$level]) {
+      return;
+    }
+    if ($this->options['logFormat'] == 'json') {
+      $message = $this->formatJsonMessages($level, $message, $context);
+    } elseif ($this->options['logFormat'] == 'bash') {
+      $message = $this->formatBashMessages($level, $message, $context);
+    } else {
+      $message = $this->formatMessages($level, $message, $context);
```

Comment:
I left all these parameters in so that these function's signatures would match the original version. I am not sure whether it is smarter to have it match for consistency and clarity, or to pass in a $parts array. Thoughts?